packages/engine-server/src/drivers/file/storev2.ts

```diff
@@ -556,12 +556,25 @@ export class FileStorage implements DStore {
     const {
       notes,
       cacheUpdates,
+      fileMetaDict,
       errors: parseErrors,
     } = await new NoteParser({
       store: this,
       cache,
       logger: this.logger,
     }).parseFiles(noteFiles, vault);
+    const seenFiles = new Set(
```

Comment:
instead of constructing it from scratch, can we build this set in `getAllFiles` and return it so we're not iterating over all notes twice?

Reply:
We call `fs.readdir` which returns an array. Converting to set will require iterating twice. However, I can reuse the data from `getAllFiles` so it makes it more readable